openpgp/read.go

```diff
@@ -436,18 +428,14 @@ func CheckDetachedSignatureAndHash(keyring KeyRing, signed, signature io.Reader,
 	}
 
 	for _, key := range keys {
-		switch sig := p.(type) {
-		case *packet.Signature:
-			err = key.PublicKey.VerifySignature(h, sig)
-			if err == nil && sig.SigExpired(config.Now()) {
-				err = errors.ErrSignatureExpired
-			}
-		case *packet.SignatureV3:
-			err = key.PublicKey.VerifySignatureV3(h, sig)
-		default:
-			panic("unreachable")
+		sig, ok := p.(*packet.Signature)
+		if !ok {
+			return nil, errors.StructuralError("non signature packet found")
+		}
```

Comment:
If you add a `var sig *packet.Signature` to the beginning of this function, you don't have to cast `p` again here